classes/ProductSale.php

```diff
@@ -234,29 +234,31 @@ public static function getBestSalesLight($idLang, $pageNumber = 0, $nbProducts =
      * Add Product sale.
      *
      * @param int $productId Product ID
+     * @param int $idShop Shop ID
      * @param int $qty Quantity
      *
      * @return bool Indicates whether the sale was successfully added
      */
-    public static function addProductSale($productId, $qty = 1)
+    public static function addProductSale($productId, $idShop, $qty = 1)
     {
         return Db::getInstance()->execute('
 			INSERT INTO ' . _DB_PREFIX_ . 'product_sale
-			(`id_product`, `quantity`, `sale_nbr`, `date_upd`)
-			VALUES (' . (int) $productId . ', ' . (int) $qty . ', 1, NOW())
+			(`id_product`, `id_shop`, `quantity`, `sale_nbr`, `date_upd`)
+			VALUES (' . (int) $productId . ', ' . (int) $idShop . ', ' . (int) $qty . ', 1, NOW())
 			ON DUPLICATE KEY UPDATE `quantity` = `quantity` + ' . (int) $qty . ', `sale_nbr` = `sale_nbr` + 1, `date_upd` = NOW()');
     }
 
     /**
      * Get number of sales.
      *
      * @param int $idProduct Product ID
+     * @param int $idShop Shop ID
      *
      * @return int Number of sales for the given Product
      */
-    public static function getNbrSales($idProduct)
+    public static function getNbrSales($idProduct, $idShop)
```

Comment:
Same suggestion for all these new parameters, you should leave them as optional and use the shop context to get fallback values